types/react-redux/react-redux-tests.tsx

```diff
@@ -102,23 +102,79 @@ function MapDispatch() {
 }
 
 function MapDispatchWithThunkActionCreators() {
-    class TestComponent extends React.Component<{
-        foo: string,
-        onClick(): void,
-        thunkAction(): Promise<void>
-    }> {}
-
-    const mapDispatchToProps = () => ({
-        onClick: () => {},
-        thunkAction: () => async () => {}
+    const simpleAction = (payload: boolean) => ({
+        type: 'SIMPLE_ACTION',
+        payload,
     });
-
-    const Test = connect(
-        null,
-        mapDispatchToProps,
+    const thunkAction = (param1: number, param2: string) => (
+        async (dispatch: Dispatch, { foo }: OwnProps) => {
+            return foo;
+        }
+    );
+    interface OwnProps {
+        foo: string;
+    }
+    interface TestComponentProps extends OwnProps {
+        simpleAction: typeof simpleAction;
+        thunkAction(param1: number, param2: string): Promise<string>;
```

Comment:
It could be nice to have a simply-named type operator that you can use to transform the types yourself.

This is how I generally declare my components' inputs:

```ts
interface OwnProps {
  // ...
}

interface ReduxProps {
  // things from mapStateToProps
}

function mapStateToProps(state: State /* if required: , ownProps: OwnProps */): ReduxProps {
  return { /* ... */ }
}

const mapDispatchToProps = {
  actionCreator1,
  actionCreator2
}

type Props = OwnProps & ReduxProps & typeof mapDispatchToProps
```

The request is for something like `type ResolveThunks<T> = ...` that you could then use like `ResolveThunks<typeof mapDispatchToProps>`.

Reply:
(oh, you have `WithThunkActionCreators` but would be nice to have a shorter name)

Reply:
Renamed `WithActionCreators` to `ResolveThunks`.